src/py/crankshaft/crankshaft/analysis_data_provider.py

```diff
@@ -2,18 +2,25 @@
 import plpy
 import pysal_utils as pu
 
+NULL_VALUE_ERROR = ('No usable data passed to analysis. Check your input rows '
+                    'for null values and fill in appropriately.')
+
+
+def verify_data(n_rows):
```

Comment:
If the goal is to have a more generic error checking capability, I think that `verify_data` should receive the actual rows, and perform a `len` on them itself.

That way, you could easily adjust the function to handle cases where a non-zero number of rows are returned, but something is wrong with the rows.